tests/basic/volume_ops_test.go

```diff
@@ -3730,3 +3737,132 @@ var _ = Describe("{RestartPxandRestartNode}", func() {
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 0; i < volumeCount; i++ {
+				volName := fmt.Sprintf("volduringkvdbflatmode-%d", i)
+				volId, err := Inst().V.CreateVolume(volName, 10, 1)
+				log.FailOnError(err, "Failed to create volume")
+				volList = append(volList, volId)
+			}
+			log.InfoD("Attach few volumes and let few volumes be in detached mode")
+			for i := 0; i < 5; i++ {
+				out, err := Inst().V.AttachVolume(volList[i])
+				log.FailOnError(err, "Failed to attach volume")
+				log.InfoD("Volume attached [%v]", out)
+			}
+		})
+		stepLog = "Put the Cluster in run Flat mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			kvdbNodes, err := GetAllKvdbNodes()
+			log.FailOnError(err, "Failed to get kvdb nodes")
+			if len(kvdbNodes) < 2 {
+				log.FailOnError(fmt.Errorf("KVDB nodes are less than 2"), "KVDB nodes are less than 2")
+			}
+
+			// Select two KVDB nodes
+			selectedNodes = make([]node.Node, 0)
+			for i := 0; i < 2; i++ {
+				kvdbNode, err := node.GetNodeDetailsByNodeID(kvdbNodes[i].ID)
+				log.FailOnError(err, "Failed to get kvdb node details")
+				log.InfoD("KVDB node to be stopped: %s", kvdbNode.Name)
+				selectedNodes = append(selectedNodes, kvdbNode)
+			}
+
+			// Stop the driver on the selected nodes
+			for _, kvdbNode := range selectedNodes {
+				err = Inst().V.StopDriver([]node.Node{kvdbNode}, false, nil)
+				log.FailOnError(err, "Failed to stop kvdb node")
+				log.InfoD("Stopped kvdb node: %s", kvdbNode.Name)
+				err = Inst().V.WaitDriverDownOnNode(kvdbNode)
+				log.FailOnError(err, "Failed to wait for driver to be down on node")
+			}
+		})
+		stepLog = "Try Deleting Volumes During flat mode and volumes should not be deleted"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 0; i < 5; i++ {
+				err := Inst().V.DeleteVolume(volList[i])
+				if err != nil {
+					isFlatModeEnabled = strings.Contains(err.Error(), kvdbRunFlatMode)
+					if isFlatModeEnabled {
+						log.InfoD("Volume [%v] is not deleted during flat mode", volList[i])
+					} else {
+						log.FailOnError(err, "Failed to delete volume")
+					}
+				} else {
+					log.FailOnError(fmt.Errorf("Volume [%v] is deleted during flat mode", volList[i]), "Volume should not be deleted during flat mode")
+				}
+			}
+		})
+		stepLog = "Try Attaching volumes during flat mode and volumes should not be attached"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 5; i < 10; i++ {
+				_, err := Inst().V.AttachVolume(volList[i])
+				if err != nil {
+					isFlatModeEnabled = strings.Contains(err.Error(), kvdbRunFlatMode)
+					if isFlatModeEnabled {
+						log.InfoD("Volume [%v] is not attached during flat mode", volList[i])
+					} else {
+						log.FailOnError(err, "Failed to attach volume")
+					}
+				} else {
+					log.FailOnError(fmt.Errorf("Volume [%v] is attached during flat mode", volList[i]), "Volume should not be attached during flat mode")
+				}
+			}
+		})
```

Comment:
can you deploy apps before going into run flat mode and validate them that they are running fine before making kvdb healthy again